### PR TITLE
Internet checking

### LINKED ISSUE #71: App crashes without device internet access
### Describe the bug

When the device that the app runs on has no internet access, the app crashes when it tries to fetch data. It will crash when the app first opens (if launched without internet access), or when performing a search (if internet access is lost once the app is open).

### To Reproduce

Steps to reproduce the behaviour:

1. Disconnect from the internet (e.g. turn on aeroplane mode)
2. Launch the app
3. Observe immediate crash

### Expected behaviour

- [x] The app should not crash when started without internet access
- [x] The app should not crash when searching without internet access
- [x] The app should notify the user if it failed to to retrieve data
- [x] When the app notifies the user, it should explain what went wrong (e.g. device has no internet access or try again later due to server error)

### PR DESCRIPTION
**Pull request by muljo**
closes #71

### CODE REVIEW

File: app/src/main/java/com/sthoray/allright/ui/base/ViewModelProviderFactory.kt [hunk 1]
```diff
@@ -12,6 +13,7 @@ import com.sthoray.allright.ui.search.viewmodel.SearchViewModel
  * @property appRepository the repository to access
```
Comment (scsole) [inline app/src/main/java/com/sthoray/allright/ui/base/ViewModelProviderFactory.kt:13]: Please remove this unneeded property comment

File: app/src/main/java/com/sthoray/allright/ui/main/viewmodel/MainViewModel.kt [hunk 2]
```diff
@@ -1,14 +1,21 @@
 package com.sthoray.allright.ui.main.viewmodel
 
+import android.app.Application
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
```
Comment (scsole) [inline app/src/main/java/com/sthoray/allright/ui/main/viewmodel/MainViewModel.kt:7]: Please optimise imports.

File: app/src/main/java/com/sthoray/allright/ui/search/viewmodel/SearchViewModel.kt [hunk 3]
```diff
@@ -1,14 +1,23 @@
 package com.sthoray.allright.ui.search.viewmodel
 
+import android.app.Application
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.ConnectivityManager.*
+import android.net.NetworkCapabilities
+import android.os.Build
```
Comment (scsole) [inline app/src/main/java/com/sthoray/allright/ui/search/viewmodel/SearchViewModel.kt:8]: Optimise imports

File: app/src/main/java/com/sthoray/allright/utils/Internet.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import android.os.Build
+
+/**
+ * Internet utils class
+ */
+class Internet {
+    companion object {
+        /**
+         * Checks if the application has a connection to the Internet
+         *@param viewModel The view model, used to get the connectivity service
```
Comment (scsole) [inline app/src/main/java/com/sthoray/allright/utils/Internet.kt:16]: This don't look right :)

**Comment by muljo** [comment]
@scsole fixed

**Comment by scsole** [comment]
> @scsole fixed

@muljo I see only one file changed. Did you forget to add the remaining files?

**Comment by muljo** [comment]
> 
> 
> > @scsole fixed
> 
> @muljo I see only one file changed. Did you forget to add the remaining files?

yes

**Pull request merged**